Lib/dataclasses.py

```diff
@@ -1326,10 +1347,27 @@ def _add_slots(cls, is_frozen, weakref_slot, defined_fields):
             if _update_func_cell_for__class__(member, cls, newcls):
                 break
         elif isinstance(member, property):
-            if (_update_func_cell_for__class__(member.fget, cls, newcls)
-                or _update_func_cell_for__class__(member.fset, cls, newcls)
-                or _update_func_cell_for__class__(member.fdel, cls, newcls)):
-                break
+            for f in member.fget, member.fset, member.fdel:
+                if f is None:
+                    continue
+                # unwrap once more in case function
+                # was wrapped before it became property
+                f = inspect.unwrap(f)
+                if _update_func_cell_for__class__(f, cls, newcls):
+                    break
+        elif hasattr(member, "__get__") and not inspect.ismemberdescriptor(
+            member
+        ):
+            # we don't want to inspect custom descriptors just yet
+            # there's still a chance we'll encounter a pure function
+            # or a property
```

Comment:
Please add a comment explaining why encountering a pure function or property means we don't have to inspect custom descriptors.

Reply:
Hmm, there's already an explanation above:
https://github.com/python/cpython/blob/3b65acf8d66e7e42ad25746976b55dc0bd7d71d8/Lib/dataclasses.py#L1336-L1339

Would you like me to add clarification for this case specifically?

Reply:
Ah, you're right. Maybe change the comment to "... fallback to inspecting custom descriptors if no pure function or property is found".

Reply:
Makes sense. I've added clarification in both places to be extra clear.